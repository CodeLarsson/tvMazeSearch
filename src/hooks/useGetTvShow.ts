import {useEffect, useState} from 'react';
import {TvMazeShow} from '../types/tv-maze-show';

export const useGetTvShow = (showId: number): TvMazeShow | undefined => {
  const [tvShowData, setTvShowData] = useState<TvMazeShow>();

  useEffect(() => {
    const abortController = new AbortController();

    const fetchData = async () => {
      const response = await fetch(`https://api.tvmaze.com/shows/${showId}`, {
        method: 'GET',
        signal: abortController.signal,
      });
      const data = await response.json();
      setTvShowData(data);

      return () => abortController.abort();
    };
    fetchData().catch(error => console.log(error));
  }, [showId]);

  return tvShowData;
};
