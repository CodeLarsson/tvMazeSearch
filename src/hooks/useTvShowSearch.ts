import {useEffect, useState} from 'react';
import {TvMazeShowResponse} from '../types/tv-maze-show';
import {AppSettings} from '../utils/AppSettings';

export const useTvShowSearch = () => {
  const [query, setQuery] = useState('');

  const [tvShowData, setTvShowData] = useState<TvMazeShowResponse[]>([]);

  const hits = tvShowData.length;

  useEffect(() => {
    const fetchData = async (q: string) => {
      try {
        const response = await fetch(
          `${AppSettings.apiUrl}search/shows?q=${q}`,
          {
            method: 'GET',
          },
        );

        if (!response.ok) {
          // Throw a custom error with the HTTP status code
          throw new Error(
            `Fetch error: ${response.status} ${response.statusText}`,
          );
        }

        const data = await response.json();

        // Check if the data is an array before updating the state
        if (Array.isArray(data)) {
          setTvShowData(data);
        } else {
          console.error('Invalid JSON response:', data);
        }
      } catch (error) {
        console.error(error);
      }
    };

    fetchData(query).catch(error => {
      console.error(error);
    });
  }, [query]);

  return {
    query,
    setQuery,
    tvShowData,
    hits,
  };
};
