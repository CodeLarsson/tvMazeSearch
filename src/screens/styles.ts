import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 24,
  },
  headingText: {
    fontSize: 24,
    lineHeight: 32,
    fontWeight: 'bold',
  },
  screenText: {
    fontSize: 18,
    lineHeight: 21,
    letterSpacing: 0.25,
  },
});
