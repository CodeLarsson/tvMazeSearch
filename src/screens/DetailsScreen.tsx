import React from 'react';
import {
  View,
  Text,
  SafeAreaView,
  ScrollView,
  useWindowDimensions,
  StyleSheet,
  Image,
} from 'react-native';
import {useGetTvShow} from '../hooks/useGetTvShow';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {RootStackParamList} from '../../App';

import styles from './styles';
import {useTheme} from '@react-navigation/native';
import RenderHTML from 'react-native-render-html';

type DetailsScreenProps = NativeStackScreenProps<RootStackParamList, 'Details'>;

export const DetailsScreen = ({route}: DetailsScreenProps) => {
  const showId = route.params.showId;
  const tvShow = useGetTvShow(showId);
  const {width} = useWindowDimensions();
  const {colors} = useTheme();

  const localStyle = StyleSheet.create({
    textStyle: {
      marginTop: 24,
    },
    showImage: {
      marginTop: 16,
      width: '100%',
      height: 500,
      resizeMode: 'cover',
    },
  });

  return (
    <SafeAreaView>
      <ScrollView contentInsetAdjustmentBehavior="automatic">
        <View style={styles.container}>
          <Text style={{color: colors.text, ...styles.headingText}}>
            {tvShow?.name}
          </Text>
          {tvShow?.image.medium && (
            <Image
              style={localStyle.showImage}
              source={{
                uri: tvShow?.image.medium,
              }}
            />
          )}
          <RenderHTML
            contentWidth={width}
            source={{html: tvShow?.summary ?? ''}}
          />

          <Text
            style={{
              color: colors.text,
              ...styles.screenText,
              ...localStyle.textStyle,
            }}
          >
            Genres: {tvShow?.genres?.join(', ')}
          </Text>
          <Text
            style={{
              color: colors.text,
              ...styles.screenText,
              ...localStyle.textStyle,
            }}
          >
            Rating: {tvShow?.rating?.average ?? 'N/A'}
          </Text>
          <Text
            style={{
              color: colors.text,
              ...styles.screenText,
              ...localStyle.textStyle,
            }}
          >
            Runtime: {tvShow?.runtime ?? 'N/A'}
          </Text>
          <Text
            style={{
              color: colors.text,
              ...styles.screenText,
              ...localStyle.textStyle,
            }}
          >
            Language: {tvShow?.language ?? 'N/A'}
          </Text>
          <Text
            style={{
              color: colors.text,
              ...styles.screenText,
              ...localStyle.textStyle,
            }}
          >
            Status: {tvShow?.status ?? 'N/A'}
          </Text>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};
