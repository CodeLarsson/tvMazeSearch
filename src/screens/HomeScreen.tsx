import React from 'react';
import {View, Text, ScrollView, Button, StyleSheet} from 'react-native';

import styles from './styles';
import {useTheme} from '@react-navigation/native';
import {SafeAreaView} from 'react-native-safe-area-context';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {RootStackParamList} from '../../App';

type HomeScreenProps = NativeStackScreenProps<RootStackParamList, 'Home'>;

export const HomeScreen = ({navigation}: HomeScreenProps) => {
  const {colors} = useTheme();

  const localstyles = StyleSheet.create({
    buttonRow: {
      marginTop: 24,
      flexDirection: 'row',
      gap: 16,
    },
  });

  return (
    <SafeAreaView
      style={{backgroundColor: colors.background, ...styles.container}}
    >
      <ScrollView contentInsetAdjustmentBehavior="automatic">
        <View>
          <Text style={{color: colors.text, ...styles.headingText}}>
            TV Shows
          </Text>
          <Text style={{color: colors.text, ...styles.screenText}}>
            Search for your favourite TV shows using the TV Maze API
          </Text>
          <View style={localstyles.buttonRow}>
            <Button
              title="Go to Search"
              onPress={() => {
                navigation.navigate('Search');
              }}
            />
          </View>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};
