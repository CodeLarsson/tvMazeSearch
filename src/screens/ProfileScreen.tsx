import React, {useState} from 'react';
import {View, Text, SafeAreaView, FlatList} from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {TvMazeShow} from '../types/tv-maze-show';
import {useTheme} from '@react-navigation/native';
import styles from './styles';

export const ProfileScreen = () => {
  const IDS_STORAGE_KEY = 'favouritesIds';

  const [favourites, setFavourites] = useState<TvMazeShow[]>([]);

  const readFavouriteShowsFromStorage = async (ids: string[]) => {
    if (ids.length === 0) {
      return [];
    }
    const items = await AsyncStorage.multiGet(ids);
    return items;
  };
  const getFavouriteIdsFromStorage = async (): Promise<string[] | null> => {
    const ids = await AsyncStorage.getItem(IDS_STORAGE_KEY);
    return ids ? JSON.parse(ids) : null;
  };

  const loadFavourites = async () => {
    // Stringified array of favourite ids
    const favouriteIdsFromStorage = await getFavouriteIdsFromStorage();

    if (favouriteIdsFromStorage && favouriteIdsFromStorage.length > 0) {
      const showsFromStorage = await readFavouriteShowsFromStorage(
        favouriteIdsFromStorage,
      );

      if (showsFromStorage && showsFromStorage.length > 0) {
        const parsedShows: TvMazeShow[] = showsFromStorage.map(show => {
          if (show[1]) {
            return JSON.parse(show[1]);
          }
        });
        return parsedShows;
      }
    }
  };

  loadFavourites()
    .then(favouritesToLoad => setFavourites(favouritesToLoad ?? []))
    .catch(error => console.log(error));

  const colors = useTheme().colors;

  return (
    <SafeAreaView
      style={{backgroundColor: colors.background, ...styles.container}}
    >
      <FlatList
        ListHeaderComponent={
          <>
            <View>
              <Text style={{color: colors.text, ...styles.headingText}}>
                Profile
              </Text>
              <Text style={{color: colors.text, ...styles.screenText}}>
                Here is a list of your favourite TV shows
              </Text>
            </View>
          </>
        }
        data={favourites}
        renderItem={({item}) => (
          <View>
            <Text>{item.name}</Text>
          </View>
        )}
      />
    </SafeAreaView>
  );
};
