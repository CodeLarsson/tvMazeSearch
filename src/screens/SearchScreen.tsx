import React, {useState} from 'react';
import {
  View,
  Text,
  TextInput,
  Button,
  FlatList,
  StyleSheet,
  useWindowDimensions,
} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import styles from './styles';
import {useTheme} from '@react-navigation/native';
import {useTvShowSearch} from '../hooks/useTvShowSearch';
import {SearchResultListItem} from '../components/search-result-list-item/SearchResultListItem';

type SearchScreenProps = {
  navigation: any;
};

export const SearchScreen = ({navigation}: SearchScreenProps) => {
  const {colors} = useTheme();
  const {width} = useWindowDimensions();

  const {query, setQuery, tvShowData, hits} = useTvShowSearch();

  const [queryInput, setQueryInput] = useState('');

  const handleClear = () => {
    setQueryInput('');
    setQuery('');
  };

  const handleSearch = () => {
    if (queryInput) {
      setQuery(queryInput);
    } else {
      setQuery('');
    }
  };

  const localstyles = StyleSheet.create({
    searchContainer: {
      padding: 24,
      width: width,
    },
    query: {
      fontSize: 14,
      fontWeight: '300',
      fontStyle: 'italic',
    },
    input: {
      marginTop: 24,
      height: 40,
      borderWidth: 1,
      padding: 10,
      borderColor: colors.text,
    },
    buttonRow: {
      marginTop: 24,
      marginBottom: 24,
      flexDirection: 'row',
      gap: 16,
    },
  });

  return (
    <SafeAreaView
      style={{backgroundColor: colors.background, ...styles.container}}
    >
      <FlatList
        style={localstyles.searchContainer}
        ListHeaderComponent={
          <>
            <View>
              <Text style={{color: colors.text, ...styles.headingText}}>
                Search
              </Text>
              <Text style={{color: colors.text, ...styles.screenText}}>
                Search for your favourite TV shows using the TV Maze API
              </Text>
            </View>

            <TextInput
              testID="tid-search-input"
              style={localstyles.input}
              placeholder="TV Show Name"
              placeholderTextColor={colors.text}
              value={queryInput}
              onChangeText={setQueryInput}
              onSubmitEditing={() => handleSearch()}
            />
            <View style={localstyles.buttonRow}>
              <Button
                testID="tid-search-button"
                title="Search"
                onPress={handleSearch}
                disabled={!queryInput}
              />
              <Button
                testID="tid-search-clear-button"
                title="Clear"
                onPress={handleClear}
                disabled={!queryInput}
              />
            </View>
            <View>
              {query ? (
                <Text style={localstyles.query}>
                  {hits} search results for "{query}"
                </Text>
              ) : (
                ''
              )}
            </View>
          </>
        }
        data={tvShowData}
        renderItem={({item}) => (
          <SearchResultListItem show={item.show} navigation={navigation} />
        )}
      />
    </SafeAreaView>
  );
};
