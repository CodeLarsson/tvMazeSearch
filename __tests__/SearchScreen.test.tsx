import {enableFetchMocks} from 'jest-fetch-mock';
enableFetchMocks();

import 'react-native';
import React from 'react';
import {it, describe, expect, jest, beforeEach} from '@jest/globals';

import {SearchScreen} from '../src/screens/SearchScreen';
import {
  act,
  renderHook,
  render,
  screen,
  fireEvent,
  waitFor,
} from '@testing-library/react-native';
import {useTvShowSearch} from '../src/hooks/useTvShowSearch';

describe('Test Component SearchScreen', () => {
  const props: any = {
    navigation: {
      navigate: jest.fn(),
    },
  };

  beforeEach(() => {
    fetchMock.resetMocks();
  });

  it('types a query', async () => {
    render(<SearchScreen {...props} />);

    await act(async () => {
      const input = await screen.findByTestId('tid-search-input');

      fireEvent.changeText(input, 'Breaking Bad');

      // Ensure the input value is updated
      expect(input.props.value).toBe('Breaking Bad');
    });
  });

  it('performs a search', async () => {
    render(<SearchScreen {...props} />);

    fetchMock.mockResponseOnce(
      JSON.stringify([
        {
          show: {
            id: 230,
            name: 'Breaking Bad',
          },
          score: 40,
        },
      ]),
    );

    await act(async () => {
      const input = await screen.findByTestId('tid-search-input');
      const searchButton = await screen.findByTestId('tid-search-button');
      const {result} = renderHook(() => useTvShowSearch());

      fireEvent.changeText(input, 'Breaking Bad');

      // Ensure the input value is updated
      expect(input.props.value).toBe('Breaking Bad');

      fireEvent.press(searchButton); // Update query

      await waitFor(async () => {
        expect(result.current.query).toBe('Breaking Bad');
        expect(result.current.hits).toBe(1);
        expect(result.current.tvShowData).toEqual([
          {
            show: {
              id: 230,
              name: 'Breaking Bad',
            },
            score: 40,
          },
        ]);
      });
    });
  });
});
