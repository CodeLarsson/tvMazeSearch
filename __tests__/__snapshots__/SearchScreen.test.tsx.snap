// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchScreen renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgb(242, 242, 242)",
      "flex": 1,
      "justifyContent": "center",
      "padding": 24,
    }
  }
>
  <RCTScrollView
    ListHeaderComponent={
      <React.Fragment>
        <View>
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "fontSize": 24,
                "fontWeight": "bold",
                "lineHeight": 32,
              }
            }
          >
            Search
          </Text>
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "fontSize": 18,
                "letterSpacing": 0.25,
                "lineHeight": 21,
              }
            }
          >
            Search for your favourite TV shows using the TV Maze API
          </Text>
        </View>
        <TextInput
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="TV Show Name"
          placeholderTextColor="rgb(28, 28, 30)"
          style={
            {
              "borderColor": "rgb(28, 28, 30)",
              "borderWidth": 1,
              "height": 40,
              "marginTop": 24,
              "padding": 10,
            }
          }
          testID="tid-search-input"
          value=""
        />
        <View
          style={
            {
              "flexDirection": "row",
              "gap": 16,
              "marginBottom": 24,
              "marginTop": 24,
            }
          }
        >
          <Button
            disabled={true}
            onPress={[Function]}
            testID="tid-search-button"
            title="Search"
          />
          <Button
            disabled={true}
            onPress={[Function]}
            testID="tid-search-clear-button"
            title="Clear"
          />
        </View>
        <View>
          
        </View>
      </React.Fragment>
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    style={
      {
        "padding": 24,
        "width": 750,
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        collapsable={false}
        onLayout={[Function]}
      >
        <View>
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "fontSize": 24,
                "fontWeight": "bold",
                "lineHeight": 32,
              }
            }
          >
            Search
          </Text>
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "fontSize": 18,
                "letterSpacing": 0.25,
                "lineHeight": 21,
              }
            }
          >
            Search for your favourite TV shows using the TV Maze API
          </Text>
        </View>
        <TextInput
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="TV Show Name"
          placeholderTextColor="rgb(28, 28, 30)"
          style={
            {
              "borderColor": "rgb(28, 28, 30)",
              "borderWidth": 1,
              "height": 40,
              "marginTop": 24,
              "padding": 10,
            }
          }
          testID="tid-search-input"
          value=""
        />
        <View
          style={
            {
              "flexDirection": "row",
              "gap": 16,
              "marginBottom": 24,
              "marginTop": 24,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="tid-search-button"
          >
            <View
              style={
                [
                  {},
                  {},
                ]
              }
            >
              <Text
                disabled={true}
                style={
                  [
                    {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    {
                      "color": "#cdcdcd",
                    },
                  ]
                }
              >
                Search
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="tid-search-clear-button"
          >
            <View
              style={
                [
                  {},
                  {},
                ]
              }
            >
              <Text
                disabled={true}
                style={
                  [
                    {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    {
                      "color": "#cdcdcd",
                    },
                  ]
                }
              >
                Clear
              </Text>
            </View>
          </View>
        </View>
        <View />
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;

exports[`Test Component SearchScreen Renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgb(242, 242, 242)",
      "flex": 1,
      "justifyContent": "center",
      "padding": 24,
    }
  }
>
  <RCTScrollView
    ListHeaderComponent={
      <React.Fragment>
        <View>
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "fontSize": 24,
                "fontWeight": "bold",
                "lineHeight": 32,
              }
            }
          >
            Search
          </Text>
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "fontSize": 18,
                "letterSpacing": 0.25,
                "lineHeight": 21,
              }
            }
          >
            Search for your favourite TV shows using the TV Maze API
          </Text>
        </View>
        <TextInput
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="TV Show Name"
          placeholderTextColor="rgb(28, 28, 30)"
          style={
            {
              "borderColor": "rgb(28, 28, 30)",
              "borderWidth": 1,
              "height": 40,
              "marginTop": 24,
              "padding": 10,
            }
          }
          testID="tid-search-input"
          value=""
        />
        <View
          style={
            {
              "flexDirection": "row",
              "gap": 16,
              "marginBottom": 24,
              "marginTop": 24,
            }
          }
        >
          <Button
            disabled={true}
            onPress={[Function]}
            testID="tid-search-button"
            title="Search"
          />
          <Button
            disabled={true}
            onPress={[Function]}
            testID="tid-search-clear-button"
            title="Clear"
          />
        </View>
        <View>
          
        </View>
      </React.Fragment>
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    style={
      {
        "padding": 24,
        "width": 750,
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        collapsable={false}
        onLayout={[Function]}
      >
        <View>
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "fontSize": 24,
                "fontWeight": "bold",
                "lineHeight": 32,
              }
            }
          >
            Search
          </Text>
          <Text
            style={
              {
                "color": "rgb(28, 28, 30)",
                "fontSize": 18,
                "letterSpacing": 0.25,
                "lineHeight": 21,
              }
            }
          >
            Search for your favourite TV shows using the TV Maze API
          </Text>
        </View>
        <TextInput
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="TV Show Name"
          placeholderTextColor="rgb(28, 28, 30)"
          style={
            {
              "borderColor": "rgb(28, 28, 30)",
              "borderWidth": 1,
              "height": 40,
              "marginTop": 24,
              "padding": 10,
            }
          }
          testID="tid-search-input"
          value=""
        />
        <View
          style={
            {
              "flexDirection": "row",
              "gap": 16,
              "marginBottom": 24,
              "marginTop": 24,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="tid-search-button"
          >
            <View
              style={
                [
                  {},
                  {},
                ]
              }
            >
              <Text
                disabled={true}
                style={
                  [
                    {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    {
                      "color": "#cdcdcd",
                    },
                  ]
                }
              >
                Search
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="tid-search-clear-button"
          >
            <View
              style={
                [
                  {},
                  {},
                ]
              }
            >
              <Text
                disabled={true}
                style={
                  [
                    {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    {
                      "color": "#cdcdcd",
                    },
                  ]
                }
              >
                Clear
              </Text>
            </View>
          </View>
        </View>
        <View />
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
